backtrace-library/src/main/java/backtraceio/library/BacktraceDatabase.java

```diff
@@ -145,7 +147,13 @@ public Boolean setupNativeIntegration(BacktraceBase client, BacktraceCredentials
         String[] values = crashpadAttributes.attributes.values().toArray(new String[0]);
 
         // Paths to Crashpad attachments
-        String[] attachmentPaths = new String[] {this.breadcrumbs.getBreadcrumbLogPath()};
+        List<String> attachmentPaths = new ArrayList<String>();
```

Comment:
Do we need to create a list? We cast it anyway to array later, so we will avoid unnecessary cast later and make code simpler with just array. We also know array size because we know how many attachments we store + additional breadcrumbs file.

Reply:
If we use a Java String Array here, we get some pretty messy hard to maintain code:
`        String[] breadcrumbAttachmentPath = {this.breadcrumbs.getBreadcrumbLogPath()};
        String[] clientAttachmentPaths = (String[]) client.attachments.toArray();
        String[] attachmentPaths = new String[breadcrumbAttachmentPath.length + clientAttachmentPaths.length];
        System.arraycopy(breadcrumbAttachmentPath, 0, attachmentPaths, 0, breadcrumbAttachmentPath.length);
        System.arraycopy(clientAttachmentPaths, 0, attachmentPaths, breadcrumbAttachmentPath.length, clientAttachmentPaths.length);`
        
In general we prefer a List<String> in Java wherever possible, to avoid messy code like the above, see: https://stackoverflow.com/questions/2843366/how-to-add-new-elements-to-an-array

But we prefer a String Array for native interface, so best to work with a List<String> while we're in Java and convert to String array when we want to go to native code.

Reply:
Thanks for sharing the example from Slack @konraddysput , we are now using String[] here!